src/scripts/setup_external_storage/setup_external_storage.py

```diff
@@ -0,0 +1,61 @@
+"""
+Create an STS-enabled folder on Synapse over an S3 location.
+"""
+
+import os
+import json
+import argparse
+
+import synapseclient
+
+
+def read_args():
+    parser = argparse.ArgumentParser(
+        description="Create an STS-enabled folder on Synapse over an S3 location."
+    )
+    parser.add_argument(
+        "--synapse-parent", help="Synapse ID of the parent folder/project"
+    )
+    parser.add_argument(
+        "--synapse-folder-name",
+        help=(
+            "Name of the Synapse folder to use with the external " "storage location."
+        ),
+    )
+    parser.add_argument("--s3-bucket", help="S3 bucket name")
+    parser.add_argument(
+        "--s3-key",
+        default=None,
+        help="S3 key to serve as the root. `/` (root) by default.",
+    )
+    parser.add_argument(
+        "--sts-enabled",
+        action="store_true",
+        help="Whether this storage location should be STS enabled",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = read_args()
+    syn = synapseclient.login()
+    storage_location = syn.create_s3_storage_location(
```

Comment:
Did you copy this script from BridgeDownstream?  I need to add it to my TODOs to create an EPIC for creating a RECOVER / BridgeDownstream utility package.  

1. should this function be returned as a dict in the python client?
2. Will having this command as a cli as part of the client remove the need for this script?

Reply:
Yes this is copied from here: [src/scripts/setup_external_storage/setup_external_storage.py](https://github.com/Sage-Bionetworks/BridgeDownstream/blob/main/src/scripts/setup_external_storage/setup_external_storage.py) except we don't copy any bucket policy (done in S3 bucket template), create any owner.txt (done in S3 bucket template) or cors rules anymore

Reply:
1. I'm looking at the current use case for this function `create_s3_storage_location` and it's not really being used elsewhere other than BridgeDownstream, here (for just this one script) and the python synapse client itself. I don't think it needs to be returned as a dict except for the benefit of just guaranteeing that should the order of the items in the dict change, it's fine since we are pulling them out by the key?
2. What would that look like?

Reply:
1. I think thats fine - I don't think the order of the keys will change because tests will fail if the order changed on the client.
2. When you install the synapsePythonClient, a command line client is installed as well.  So... something like this in the terminal

    ```
     synapse create-s3-storage-location ...
     ```

Reply:
>     2. When you install the synapsePythonClient, a command line client is installed as well.  So... something like this in the terminal
>        ```
>         synapse create-s3-storage-location ...
>        ```
Oh I see. Does the cli command `synapse login` allow for logging in with an `authToken` instead of cached credentials? But, yes I can see that it would eliminate the use for this script if we didn't plan to expand this script to anything else other than this.

Reply:
Got it. I will add this to the backlog of tickets to potentially create and track the usecase.